src/sql/src/plan/statement.rs

```diff
@@ -444,15 +458,10 @@ fn finish_show_where(
 
 fn handle_show_databases(
     scx: &StatementContext,
-    ShowDatabasesStatement { filter }: ShowDatabasesStatement,
+    ShowDatabasesStatement { query, filter }: ShowDatabasesStatement,
 ) -> Result<Plan, anyhow::Error> {
-    let rows = scx
-        .catalog
-        .list_databases()
-        .map(|database| vec![Datum::from(database)])
-        .collect();
-
-    finish_show_where(scx, filter, rows, &SHOW_DATABASES_DESC)
+    let (expr, _, _) = query::plan_root_query(scx, *query, QueryLifetime::OneShot)?;
```

Comment:
Following on from above comment, if you can stick the `filter` here onto the `Select` that you synthesize, then you should be able to let `plan_root_query` handle the filtering too. In fact, I think you'd probably just want to call `handle_select` on the synthesized `Select`!

Sorry—maybe I've been a bit obtuse in describing the motivations!—but when everything has been ported over to `mz_catalog` views, we should be able to ditch `plan_show_where` and `finish_show_where`, and just let the normal `handle_select` code path handle everything!